packages/@aws-cdk/aws-servicecatalogappregistry/lib/application-associator.ts

```diff
@@ -17,58 +17,86 @@ export interface ApplicationAssociatorProps {
 }
 
 /**
- * An AppRegistry construct to automatically create an application with the given name and description.
+ * An AppRegistry construct to automatically create applications with the given name and description
+ * in the desired regions.
  *
  * The application name must be unique at the account level and it's immutable.
+ * Only one application per region should be specified because a stack can only be associated with
+ * one application in a region.
  * This construct will automatically associate all stacks in the given scope, however
  * in case of a `Pipeline` stack, stage underneath the pipeline will not automatically be associated and
  * needs to be associated separately.
  *
+ * If cross region stack is detected, then this construct will give a warning and skip that association.
+ * Only stacks and an application within the same region will be associated.
+ *
  * If cross account stack is detected, then this construct will automatically share the application to consumer accounts.
  * Cross account feature will only work for non environment agnostic stacks.
  */
 export class ApplicationAssociator extends Construct {
   /**
-   * Created or imported application.
+   * Map of region to application. An application can be one that is created or imported.
+   * Because a stack can only be associated with one application in a region,
+   * one region will map to only one application.
```

Comment:
```suggestion
   * Maps one region to exactly one application.
   * The mapping is 1:1 because a stack can only be associated
   * with one application in a region.
```